### PR TITLE
Cargo shuttle now doesn't sell people

### PR DESCRIPTION
**Pull request by LuisDSG**
'Edited the CargoShuttle script and separated its item scanner from the "UnloadCargo" function before attaching it to a public "SearchForObjectsOnShuttle" function that returns a Transform Object
-Added a "LifeformChecker" function in CargoManager that looks for any players or animals on the shuttle.
-Modified the "CallShuttle" function on CargoManager so that, using "LifeformChecker", it avoids flying off from the station with a player or animal.

### Purpose
Fixes the issue that allowed the cargo shuttle to fly off from the station with characters on board and erase them afterwards by preventing the shuttle from flying as long as any object that could be alive is aboard.
### Notes:
None

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/CargoManager.cs [hunk 1]
```diff
@@ -59,6 +59,23 @@ private void OnDisable()
 		SceneManager.activeSceneChanged -= OnRoundRestart;
 	}
 
+	/// <summary>
+	/// Checks for any Lifeforms (dead or alive) that might be aboard the shuttle
+	/// </summary>
+	/// <returns>true if a lifeform is found, false if none is found</returns>
+	bool LifeformChecker()
```
Comment (ReiderJack) [inline UnityProject/Assets/Scripts/Managers/CargoManager.cs]: Good rule is to call methods with a verb. In this case for example: `CheckLifeforms`

Comment (ReiderJack) [reply UnityProject/Assets/Scripts/Managers/CargoManager.cs]: Also we specify access modifiers. At least doing it lately. So please add `private`

File: UnityProject/Assets/Scripts/Managers/CargoManager.cs [hunk 2]
```diff
@@ -59,6 +59,23 @@ private void OnDisable()
...
+	{
+		Transform ObjectHolder = CargoShuttle.Instance.SearchForObjectsOnShuttle();
+		for (int i = 0; i < ObjectHolder.childCount; i++)
+		{
+			if (ObjectHolder.GetChild(i).GetComponent<LivingHealthBehaviour>() != null)
```
Comment (ReiderJack) [inline UnityProject/Assets/Scripts/Managers/CargoManager.cs]: This is really expensive way to check if an object is a player. Did you try to check by layers?
<https://docs.unity3d.com/Manual/Layers.html> 
There is a Player layer.

Comment (ReiderJack) [reply UnityProject/Assets/Scripts/Managers/CargoManager.cs]: `if(((1<<gameObject.layer) & layers) == 0)` return `true`;
Where layers has Player layer. 
<https://docs.unity3d.com/ScriptReference/LayerMask.html>

Comment (ReiderJack) [reply UnityProject/Assets/Scripts/Managers/CargoManager.cs]: You can do ` LayerMask.GetMask("Player")`, but cache it before iterating in `for` to not call it everytime.

**Review by ReiderJack — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Managers/CargoManager.cs (hunk 2 above)
Comment (ReiderJack) [reply UnityProject/Assets/Scripts/Managers/CargoManager.cs]: Also you can iterate without using `for`
```
foreach (Transform child in transform)
{
  //child is your child transform
}
```

**Comment by corp-0** [comment]
@ReiderJack it is mimicking ss13 behavior. It doesn’t just check players but anything that has/had life.

**Comment by corp-0** [comment]
That being said, you can check player and NPC layer to get the same result.

Edit: and dead layer too

**Comment by ReiderJack** [comment]
But Ian has value and he can be sold

**Comment by corp-0** [comment]
In OG SS13 you can’t sell Ian not even dead mobs/players

**Comment by ReiderJack** [comment]
I see the point in not selling alive players, but don't see one for mobs or dead bodies.

**Comment by corp-0** [comment]
Hiding bodies would be way too easy if you can just sell them.

**Comment by AutumnNova** [comment]
not to mention it deletes the player, meaning they cannot be revived with admin intervention at all

**Comment by ReiderJack** [comment]
Just use layers to determine if there is a living thing on a matrix, not `GetComponent`

File: UnityProject/Assets/Scripts/Managers/CargoManager.cs [hunk 3]
```diff
@@ -59,6 +59,23 @@ private void OnDisable()
...
+	/// <summary>
+	/// Checks for any Lifeforms (dead or alive) that might be aboard the shuttle
+	/// </summary>
+	/// <returns>true if a lifeform is found, false if none is found</returns>
+	bool CheckLifeforms()
+	{
+		Transform ObjectHolder = CargoShuttle.Instance.SearchForObjectsOnShuttle();
+		foreach (Transform child in ObjectHolder)
+		{
+			if (child.transform.gameObject.layer == 12 || child.transform.gameObject.layer == 8)
```
Comment (ReiderJack) [inline UnityProject/Assets/Scripts/Managers/CargoManager.cs]: Can't you just do `child.gameObject.layer`? Because `child` is already `transform`.

Comment (ReiderJack) [reply UnityProject/Assets/Scripts/Managers/CargoManager.cs]: Cache `LayerMask` before `foreach`
`LayerMask layersToCheck = LayerMask.GetMask("Player", "NPC")`

Then do the check `if((1<<child.gameObject.laer & layersToCheck) == 0) continue;`
<https://answers.unity.com/questions/422472/how-can-i-compare-colliders-layer-to-my-layermask.html>

**Review by ReiderJack — APPROVED** [review]

**Pull request merged**